### PR TITLE
Add Orca type hint

### PR DESCRIPTION
**Pull request by hkvision**

### CODE REVIEW

File: python/orca/src/bigdl/orca/inference/inference_model.py [hunk 1]
```diff
@@ -145,7 +173,8 @@ def load_torch(self, model_path):
         callZooFunc(self.bigdl_type, "inferenceModelLoadPytorch",
                     self.value, bys.getvalue())
 
-    def predict(self, inputs):
+    # TODO: Add UT for different inputs
+    def predict(self, inputs: Union[ndarray, List[ndarray], JTensor, List[JTensor]]):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/inference/inference_model.py:177]: surround ndarray with apostrophes: `'ndarray'`

File: python/orca/src/bigdl/orca/learn/metrics.py [hunk 2]
```diff
@@ -54,11 +74,11 @@ def get_mxnet_metric(self):
         invalidInputError(False, "not implemented")
 
     @abstractmethod
-    def get_name(self):
+    def get_name(self) -> str:
         pass
 
     @staticmethod
-    def convert_metrics_list(metrics, backend="bigdl"):
+    def convert_metrics_list(metrics, backend: str = "bigdl"):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/learn/metrics.py:81]: What's the type of `metrics`?

File: python/orca/src/bigdl/orca/learn/metrics.py [hunk 3]
```diff
@@ -77,7 +97,7 @@ def convert_metrics_list(metrics, backend="bigdl"):
         return metric_impls
 
     @staticmethod
-    def convert_metrics_dict(metrics, backend="bigdl"):
+    def convert_metrics_dict(metrics, backend: str = "bigdl"):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/learn/metrics.py:100]: What's the type of metrics?

File: python/orca/src/bigdl/orca/ray/ray_on_spark_context.py [hunk 4]
```diff
@@ -527,22 +546,22 @@ def purge(self):
...
+    def _update_extra_params(self, extra_params: Optional[Dict[str, str]]) -> Dict[str, str]:
         kwargs = {}
         if extra_params is not None:
             for k, v in extra_params.items():
                 kw = k.replace("-", "_")
                 kwargs[kw] = v
         return kwargs
 
-    def init(self, driver_cores=0):
+    def init(self, driver_cores: int = 0):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/ray/ray_on_spark_context.py:564]: return a dict[str, str]?

File: python/orca/src/bigdl/orca/ray/raycontext.py [hunk 5]
```diff
@@ -59,7 +61,7 @@ def __init__(self,
 
         OrcaRayContext._active_ray_context = self
 
-    def init(self, driver_cores=0):
+    def init(self, driver_cores: int = 0):
```
Comment (leonardozcm) [inline python/orca/src/bigdl/orca/ray/raycontext.py:64]: return a ray._private.worker.BaseContext?

**Review by leonardozcm — APPROVED** [review]

File: python/orca/src/bigdl/orca/learn/metrics.py (hunk 2 above)
Comment (hkvision) [reply python/orca/src/bigdl/orca/learn/metrics.py:81]: quite complicated, not target to add type here, this method is not for users

File: python/orca/src/bigdl/orca/ray/ray_on_spark_context.py (hunk 4 above)
Comment (hkvision) [reply python/orca/src/bigdl/orca/ray/ray_on_spark_context.py:564]: Not necessarily. The return from ray is complicated and ray itself doesn't hint the return type. So we also ignore here.

File: python/orca/src/bigdl/orca/inference/inference_model.py [hunk 6]
```diff
@@ -106,9 +129,14 @@ def load_tensorflow(self, model_path, model_type="frozenModel", intra_op_paralle
                     self.value, model_path, model_type, intra_op_parallelism_threads,
                     inter_op_parallelism_threads, use_per_session_threads)
 
-    def load_tensorflow(self, model_path, model_type,
-                        inputs, outputs, intra_op_parallelism_threads=1,
-                        inter_op_parallelism_threads=1, use_per_session_threads=True):
+    # TODO: Add UT
+    def load_tensorflow_graph(self, model_path: str,
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/inference/inference_model.py:133]: Can't have two methods with the same name for mypy. Thus changing this name

**Pull request merged**